### PR TITLE
Windows ASLR fixes; detect Slackware sysctl

### PR DESCRIPTION
**Pull request by slipher**

### CODE REVIEW

File: src/engine/framework/VirtualMachine.cpp [hunk 1]
```diff
@@ -56,6 +56,32 @@ SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
...
+	static const bool problemDetected = [] {
+		try {
+			FS::File file = FS::RawPath::OpenRead("/proc/sys/vm/mmap_min_addr");
+			char buf[20];
+			buf[file.Read(&buf, sizeof(buf) - 1)] = '\0';
+			Log::Notice("minaddr buf %s", buf);
+			int minAddr;
+			return sscanf(buf, "%d", &minAddr) == 1  && minAddr > 0x10000;
+		} catch (std::system_error&) {
+			Log::Notice("BUBUBUBUB");
```
Comment (illwieckz) [inline src/engine/framework/VirtualMachine.cpp:72]: is that wip code / leftover?

Comment (slipher) [reply src/engine/framework/VirtualMachine.cpp]: Fixed.

**Comment by illwieckz** [comment]
Why not:

```c++
	if (problemDetected) {
		Sys::Error("Your system is configured with a sysctl option which makes the game unable to run.\n"
		           "To permanently fix the configuration, run the following commands:\n"
		           "\n"
		           "    echo vm.mmap_min_addr=65536 | sudo dd of=/etc/sysctl.d/daemonengine-nacl-mmap.conf\n"
		           "    sudo /sbin/sysctl --system");
	}
```

That may look better (message box was a bit wide on my end), also we may just call `sysctl --system` that will just reload everything including the new file. This would also make the message more game agnostic (it's expected the game name to be already on window title anyway), same for the file. Also I noticed it is `vm.mmap` so it's better to do `vm-mmap` or just `mmap`, but `vmmap` does not really make sense.

Before:

[mmap message](https://dl.illwieckz.net/b/daemon/bugs/mmap-message/20201222-110050-000.unvanquished-mmap-message.png)

[mmap message](https://dl.illwieckz.net/b/daemon/bugs/mmap-message/20201222-110155-000.unvanquished-mmap-message.png)

After:

[mmap message](https://dl.illwieckz.net/b/daemon/bugs/mmap-message/20201222-112151-000.unvanquished-mmap-message.png)

[mmap message](https://dl.illwieckz.net/b/daemon/bugs/mmap-message/20201222-112201-000.unvanquished-mmap-message.png)

**Comment by slipher** [comment]
I have applied the suggestions.

**Pull request merged**